modules/ding_smartbanner/ding_smartbanner.module

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * @file
+ * Smart banner module file.
+ */
+
+define('DING_SMARTBANNER_MODULE_PATH', drupal_get_path('module', 'ding_smartbanner'));
+
+/**
+ * Implements hook_menu().
+ */
+function ding_smartbanner_menu() {
+  $item = array();
+
+  $item['admin/config/smartbanner'] = array(
+    'title' => 'Smart banner',
+    'description' => 'Manage smart banner configuration.',
+    'position' => 'left',
+    'weight' => 20,
+    'page callback' => 'system_admin_menu_block_page',
+    'access arguments' => array('access administration pages'),
+    'file' => 'system.admin.inc',
+    'file path' => drupal_get_path('module', 'system'),
+  );
+
+  $item['admin/config/smartbanner/settings'] = array(
+    'title' => 'Smart banner',
+    'description' => 'Manage smart banner configuration.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_smartbanner_settings_form'),
+    'access arguments' => array('access administration pages'),
+    'file' => 'ding_smartbanner.admin.inc',
+    'type' => MENU_NORMAL_ITEM,
+  );
+
+  return $item;
+}
+
+/**
+ * Implements hook_libraries_info().
+ */
+function ding_smartbanner_libraries_info() {
+  $libraries = array();
+
+  $libraries['smart-app-banner'] = array(
+    'name' => 'Smart App Banner',
+    'vendor url' => 'https://github.com/kudago/smart-app-banner',
+    'download url' => 'https://github.com/kudago/smart-app-banner/archive/v1.3.0.zip',
+    'version callback' => 'ding_smartbanner_short_circuit_version',
+    'version arguments' => array(
+      'file' => 'smart-app-banner.js',
+      'pattern' => '/v(\d+\.+\d+\.+\d+)/',
+    ),
+    'files' => array(
+      'js' => array('smart-app-banner.js'),
+      'css' => array('smart-app-banner.css'),
+    ),
+  );
+
+  return $libraries;
+}
+
+/**
+ * Short-circuit the version argument for libraries_info().
+ */
+function ding_smartbanner_short_circuit_version() {
+  return TRUE;
+}
+
+/**
+ * Implements hook_init().
+ */
+function ding_smartbanner_init() {
+  $configs = variable_get('smartbanner_config', array());
+
+  drupal_add_css(libraries_get_path('smart-app-banner') . '/smart-app-banner.css');
+  drupal_add_js(libraries_get_path('smart-app-banner') . '/smart-app-banner.js', array('type' => 'file', 'scope' => 'footer'));
+  drupal_add_js(DING_SMARTBANNER_MODULE_PATH . '/js/ding_smartbanner.js', array('type' => 'file', 'scope' => 'footer'));
+
+  $icon_path = DING_SMARTBANNER_MODULE_PATH . '/icon.png';
```

Comment:
Maybe move the default provided icon into "image" subfolder inside the module.

Reply:
The default icon commit with this module is that really the library app's default icon?